### ISSUE TITLE
false positives on inert_defer

### ISSUE THREAD

**Issue by jjetrifork** [realm/SwiftLint#2500]
### New Issue Checklist

- [ x] Updated SwiftLint to the latest version
- [ x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The `inert_defer` rule yields false positives if  wrapped in an `if` statement as the last statement when there are more statements following the `if` statement.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths 
<redacted>
<redacted>.swift:500:13: warning: Inert Defer Violation: If defer is at the end of its parent scope, it will be executed right where it is anyway. (inert_defer)
<redacted>
Done linting! Found 1 violation, 0 serious in 115 files.
```

### Environment

Version 0.29.1
Installed thru HomeBrew

```yml
disabled_rules: # rule identifiers to exclude from running
  - line_length

excluded: # paths to ignore during linting. Takes precedence over `included`.
  - Pods
  - Podfile
  - Podfile.lock

identifier_name:
  excluded:
    - id
    - io
    - on
    - of
```

### Example
```swift
// This triggers a violation:
    func test(bool: Bool) {
        if bool {
            print("ending...")
            defer { print("Ended") }
        }
        print("I am before `defer`")
    }
```

**Comment by marcelofabri** [comment]
This is the output I get:

```
$ swift file.swift 
ending...
Ended
I am before `defer`
```

Which is why the rule exists, so I'm not sure how this is a false positive.

**Comment by jjetrifork** [comment]
I had the incorrect assumption that defer would run when the current scope ends *and* that an if statement does not create a scope. The current rule correctly triggers a violation because what I should be doing is:

    func test(bool: Bool) {
        if bool {
            print("ending...")
        }
        defer {
            if bool {
                print("Ended")
            }
        }
        print("I am before `defer`")
    }
